drivers/fuel_gauge/max17048/emul_max17048.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2023, Alvaro Garcia <maxpowel@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Emulator for max17048 fuel gauge
+ */
+
+
+#define DT_DRV_COMPAT maxim_max17048
+
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(maxim_max17048);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i2c_emul.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "max17048.h"
+
+/** Run-time data used by the emulator */
+struct max17048_emul_data {
+	uint16_t mfr_acc;
+};
+
+/** Static configuration for the emulator */
+struct max17048_emul_cfg {
+	/** I2C address of emulator */
+	uint16_t addr;
+};
+
+static int emul_max17048_reg_write(const struct emul *target, int reg, int val)
```

Comment:
Let's remove the commented out code and probably just return -EIO if we're not expecting a write